### PR TITLE
Refactor a few deploy-related classes

### PR DESCRIPTION
**Pull request by chanseokoh**
Somehow I lost all the code changes in #2411, so I restored them here.

### CODE REVIEW

**Comment by chanseokoh** [comment]
@briandealwis I got what you meant (https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/2411#discussion_r140624305). It's a separate issue, and it should be fixed in another PR. In this PR, I didn't change the original logic.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/internal/RelativeFileFieldSetter.java [hunk 1]
```diff
@@ -41,18 +40,18 @@
   private final FileDialog dialog;
   private final String[] filterExtensions;
 
-  public RelativeFileFieldSetter(Text directoryField, IPath basePath, String[] filterExtensions) {
-    this(directoryField, basePath, filterExtensions, new FileDialog(directoryField.getShell()));
+  public RelativeFileFieldSetter(Text fileField, IPath basePath, String[] filterExtensions) {
+    this(fileField, basePath, filterExtensions, new FileDialog(fileField.getShell()));
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/internal/RelativeFileFieldSetter.java]: This should use `SWT.SHEET` as a style bit.

**Comment by chanseokoh** [comment]
@briandealwis, added Javadocs. This should do it?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java [hunk 2]
```diff
@@ -223,12 +225,40 @@ private void setupProjectSelectorDataBinding() {
...
+   * Binds a {@link Text} field with a property of the {@link DeployPreferences deploy
+   * preferences model} given to the panel.
+   *
+   * Unlike {@link #setupFileFieldDataBinding}, {@code setAfterGetValidator} is always enforced
+   * regardless of the panel's validation mode.
+   *
+   * @see #setupTextFieldDataBinding
+   */
+  private void setupTextFieldDataBindingWithPermanentValidation(Text text, String modelPropertyName,
       IValidator setAfterGetValidator) {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java]: this variable would be better named as `afterGetValidator`

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java [hunk 3]
```diff
@@ -223,12 +225,40 @@ private void setupProjectSelectorDataBinding() {
         gcpProjectToProjectId, projectIdToGcpProject);
   }
 
-  private void setupTextFieldDataBinding(Control control, String modelPropertyName,
+  /**
+   * Binds a {@link Text} field with a property of the {@link DeployPreferences deploy
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java]: Why not make this `setupPossiblyUnvalidatedTextFieldDataBinding` or `OptionallyValidated`

**Pull request merged**